src/features/student/declaredPrograms/components/interactions/inputs/DeclaredProgramSourceOfInformationInput/DeclaredProgramSourceOfInformationInput.test.ts

```diff
@@ -0,0 +1,155 @@
+      const input = wrapper.findComponent({ name: 'AvInput' })
+      expect(input.exists()).toBe(true)
+    })
+
+    BddTest().then('it should have default label from i18n', () => {
+      const input = wrapper.findComponent({ name: 'AvInput' })
+      expect(input.props('label')).toBe('Provenance de l\'information')
+    })
+
+    BddTest().then('it should have default placeholder from i18n', () => {
+      const input = wrapper.findComponent({ name: 'AvInput' })
+      expect(input.props('placeholder')).toBe('Source de l\'information')
+    })
+
+    BddTest().then('it should have default prefix icon', () => {
+      const input = wrapper.findComponent({ name: 'AvInput' })
+      expect(input.props('prefixIcon')).toBe(MDI_ICONS.NEWSPAPER_VARIANT)
+    })
+
+    BddTest().then('it should have max length from config', () => {
+      const input = wrapper.findComponent({ name: 'AvInput' })
+      expect(input.props('maxlength')).toBe(DECLARED_PROGRAM_SOURCE_OF_INFORMATION_MAX_LENGTH)
+    })
+
+    BddTest().then('it should have labelVisible set to true', () => {
+      const input = wrapper.findComponent({ name: 'AvInput' })
+      expect(input.props('labelVisible')).toBe(true)
+    })
+  })
+
+  BddTest().and('the component is mounted with custom label', () => {
+    beforeEach(() => {
+      vi.clearAllMocks()
+      wrapper = mount(DeclaredProgramSourceOfInformationInput, {
+        props: {
+          label: 'Custom Label'
+        },
+        global: { stubs }
+      })
+    })
+
+    BddTest().then('it should override default label', () => {
+      const input = wrapper.findComponent({ name: 'AvInput' })
+      expect(input.props('label')).toBe('Custom Label')
+    })
+  })
+
+  BddTest().and('the component is mounted with custom prefix icon', () => {
+    beforeEach(() => {
+      vi.clearAllMocks()
+      wrapper = mount(DeclaredProgramSourceOfInformationInput, {
+        props: {
+          prefixIcon: MDI_ICONS.ATTACH_FILE
+        },
+        global: { stubs }
+      })
+    })
+
+    BddTest().then('it should override default prefix icon', () => {
+      const input = wrapper.findComponent({ name: 'AvInput' })
+      expect(input.props('prefixIcon')).toBe(MDI_ICONS.ATTACH_FILE)
+    })
+  })
+
+  BddTest().and('the component is mounted with custom placeholder', () => {
+    beforeEach(() => {
+      vi.clearAllMocks()
+      wrapper = mount(DeclaredProgramSourceOfInformationInput, {
+        props: {
+          placeholder: 'Custom Placeholder'
+        },
+        global: { stubs }
+      })
+    })
+
+    BddTest().then('it should override default placeholder', () => {
+      const input = wrapper.findComponent({ name: 'AvInput' })
+      expect(input.props('placeholder')).toBe('Custom Placeholder')
+    })
+  })
+
+  BddTest().and('the user enters text', () => {
+    beforeEach(async () => {
+      vi.clearAllMocks()
+      wrapper = mount(DeclaredProgramSourceOfInformationInput, {
+        global: { stubs }
+      })
```

Comment:
```suggestion
  BddTest().and('the user enters text', () => {
```

idem que les autres fichiers de test : and dans when et suppression du code répété, devrait être dans le when primaire (component is mounted)